src/mpi/attr/attr_impl.c

```diff
@@ -678,13 +712,17 @@ int MPIR_Type_delete_attr_impl(MPIR_Datatype * type_ptr, MPII_Keyval * keyval_pt
          * test suite says we should return the user return code.  So
          * we must not ERR_POP here. */
         mpi_errno = MPIR_Call_attr_delete(type_ptr->handle, p);
-        if (mpi_errno)
+        /* If the user delete function fails, we still remove the attribute from the list. */
+        if (mpi_errno != MPI_SUCCESS)
             goto fn_fail;
 
+        /* NOTE: it's incorrect to remove p by its parent pointer because the delete function
+         *       may have invalidated the parent pointer, e.g. by removing the parent attribute */
```

Comment:
I don't know what do you want me to fix. This same comment was in the comm code, I just copied it over here and in win.

Yes, maybe this is worth a more careful treatment.  The attribute delete function may also remove another attribute from the object. In that case, a simple way to not leave the list in an bad state is to implement the semantics I proposed elsewhere but you rejected, that is, before deleting an attribute, you first unlink the cell from the list, next to invoke the attribute delete callback. Again, you implicitly rejected such behavior, because if the delete attr callback fails, the attribute would be effectively gone forever, exactly as I wanted. I have not idea if whoever implemented these attribute management routines thought about reentrancy/recursion. I'm  not going to claim that the current implementation is wrong, but things look at least suspicious. IMHO, addressing all these issues is out of the scope of my proposed changes in this PR.

I'm not sure whether this is the intention of the original comment in comm, or the note was a clarification about why the attribute deleting was done via the auxiliary function `delete_attribute()`.

What do you want me to do exactly?

Reply:
OK, you merged. I'm still not sure what were you trying to tell me.

Reply:
> What do you want me to do exactly?

Nothing. I am just leaving some review notes and context here for future review.

Reply:
> Oh, OK, I misunderstood... Anyway, look at my comments above, I believe I may be able to break the attribute management logic easily by deleting attributes in a delete callback for another attribute.
> 
> Does ANL have any bug bounty program with juicy cash prizes 🤣 ?

No, we are overly extended in our national debt :( Is personal favor sufficient?